qiita_ware/processing_pipeline.py

```diff
@@ -83,6 +83,47 @@ def _get_qiime_minimal_mapping(prep_template, out_dir):
     return output_fps
 
 
+def _get_sample_names_by_run_prefix(prep_template):
+    """Generates a dictionary of run_prefix and sample names
+
+    Parameters
+    ----------
+    prep_template : PrepTemplate
+        The prep template from which we need to generate the minimal mapping
+
+    Returns
+    -------
+    dict
+        The paths to the qiime minimal mapping files
+
+    Raises
+    ------
+    ValueError
+        If there is more than 1 sample per run_prefix
+    """
+    from qiita_db.metadata_template import load_template_to_dataframe
+
+    # The prep templates has a QIIME mapping file, get it
+    qiime_map = load_template_to_dataframe(prep_template.qiime_map_fp,
+                                           index='#SampleID')
+
+    samples = {}
+    errors = []
+    for prefix, df in qiime_map.groupby('run_prefix'):
+        len_df = len(df)
+        if len_df != 1:
+            errors.append('%s (%d): %s' % (prefix, len_df,
```

Comment:
This is very useful, but perhaps the way to go here is to phrase this as a sentence i.e. `%s has %d samples (%s) % (prefix, len_df, ', '.join(df.index))`.

Also, note that you don't need to do `.values`, `Index` objects are already iterable and can be passed to `.join` seamlessly:

```python
In [3]: pd.DataFrame([[1, 2, 3], [4, 5, 6]], ['A', 'B'], ['r1', 'r2', 'r3'])
Out[3]: 
   r1  r2  r3
A   1   2   3
B   4   5   6

In [4]: _3.index
Out[4]: Index([u'A', u'B'], dtype='object')

In [5]: ', '.join(_3.index)
Out[5]: 'A, B'
```